### ISSUE TITLE
Add automated accessibility testing

### ISSUE THREAD

**Issue by mrlacey** [microsoft/WindowsTemplateStudio#3723]
The wizard and all generated apps should not include functionality that fails basic accessibility tests.

While automated testing on its own is not a guarantee of accessible use for all, automated tooling can find some simple, fixable issues.
Addressing this will improve the usability of the wizard and help ensure that generated apps are reflecting best practices that developers can follow.

Add tests for the following:

- [x] ~The wizard(s) and all options~
- [x] Generated UWP apps (all* option combinations)
- [x] Generated WPF apps (all* option combinations)
- [x] Address any issues found from the above.
- [x] ~Consider adding such tests as a template option that can be generated from the wizard. (These would extend the existing WinAppDriver tests)~

`*` "all" implies that all templates be tested in generated apps, not that it is necessary to test every combination of items that can be generated.

**Comment by mrlacey** [comment]
Because of the way that VSEmulator works, I cannot get automated testing of the wizard to work reliably. Will do manual checks of the wizard for accessibility.

**Comment by mrlacey** [comment]
After doing an initial pass of the UWP with an automated test I found the following.

- Issue on the **Chart** page that is easily fixed.
- Issues on the **DataGrid** page that are easily fixed.
- Issues on the **TelerikDataGrdi** page that need _further investigation_ as I believe they're due to the control.
- Issue on the **ImageGallery** page that is easily fixed.
- Issue on the **TabView** page that needs _further investigation_ as I believe is due to the control.
- Issue on the **TreeView** page that needs _further investigation_ as I believe is due to the control.
- Issues on the **WebView** page that can be ignored as due to the content of the page that is loaded by default.
- Issue on the **MediaPlayer** page that needs _further investigation_ as I believe is due to the control.
- Issues on the **Map** page. One is easily fixed, but the other needs _further investigation_ as I believe is due to the control.

progress continues....

**Comment by mrlacey** [comment]
Deciding against adding a template to do the equivalent test as being done for this as not confident in them being reliable.

**Comment by mrlacey** [comment]
For reference: https://www.access-board.gov/guidelines-and-standards/communications-and-it/about-the-ict-refresh/final-rule/single-file-version#502-interoperability-assistive-technology

**Comment by mrlacey** [comment]
Some of the accessibility-related issues are tied to the way the Symbols/Icons are used from the SampleDataService.
These Unicode values don't have appropriate accessible text.

There are 4 options for dealing with this.

1. Remove the use of symbols from what's displayed in the sample content.
2. Exclude the symbols from being visible to the automation/accessibility tools.
3. Add a new property to the `SampleOrder` object to include the symbol name that is made available to the a11y tools.
4. Add a converter (or similar) to provide a useful way of describing the symbol/icon that is made available to the a11y tools.

I'd like to opt for option 3 as this is the smallest change, doesn't limit accessible access to the content, maintains the existing UI, and doesn't add to the temporary sample content that will eventually need removing from the generated code.

Option 1 means we lose the benefit of having symbols in the sample content. As it was added to show the display of more than just text this would be a step backward.
Option 2 would be a bad example of how to treat accessibility issues.
Option 4 would mean adding additional files to the generated code that exists only to work with sample data. In turn, this would lead to more files that need to be removed from the generated code.

any objections with me moving forward with option 3?

**Comment by sibille** [comment]
Agree that option 3 is the correct one. Would using the CompanyName or OrderId make sense?

**Comment by mrlacey** [comment]
> Agree that option 3 is the correct one. Would using the CompanyName or OrderId make sense?

I've looked at using existing properties but nothing makes sense.
Because they are essentially arbitrary there isn't an equivalent other property to which they relate.
On the ContentGrid they distinguish between multiple entries for the same company.
On the DataGrid they're given their own column and so need a meaningful description if they have value in being displayed.

**Comment by mrlacey** [comment]
Issues with the Telerik RadDataGrid logged at https://github.com/telerik/UI-For-UWP/issues/466

**Comment by mrlacey** [comment]
Created related WPF issue at https://github.com/dotnet/wpf/issues/3296

**Comment by mrlacey** [comment]
These have now been added.
No impact on what ships
